contracts/test/StakeTable.t.sol

```diff
@@ -755,3 +763,340 @@ contract StakeTableUpgradeTest is Test {
+    address[] public executors = [makeAddr("executor")];
+    LightClientMock public lcMock;
+    EspToken public token;
+    StakeTableMock public stakeTable;
+    Timelock public timelock;
+    uint256 public constant INITIAL_BALANCE = 5 ether;
+    uint256 public constant ESCROW_PERIOD = 1 weeks;
+    uint256 public constant DELAY = 15 seconds;
+
+    function setUp() public {
+        tokenGrantRecipient = makeAddr("tokenGrantRecipient");
+        validator = makeAddr("validator");
+        delegator = makeAddr("delegator");
+        timelockAdmin = makeAddr("timelockAdmin");
+
+        string[] memory cmds = new string[](3);
+        cmds[0] = "diff-test";
+        cmds[1] = "mock-genesis";
+        cmds[2] = "5";
+
+        bytes memory result = vm.ffi(cmds);
+        (
+            LightClientMock.LightClientState memory state,
+            LightClientMock.StakeTableState memory stakeState
+        ) = abi.decode(result, (LightClient.LightClientState, LightClient.StakeTableState));
+        LightClientMock.LightClientState memory genesis = state;
+        LightClientMock.StakeTableState memory genesisStakeTableState = stakeState;
+
+        lcMock = new LightClientMock(genesis, genesisStakeTableState, 864000);
+
+        DeployEspTokenScript tokenDeployer = new DeployEspTokenScript();
+        (address tokenAddress,) = tokenDeployer.run(tokenGrantRecipient);
+        token = EspToken(tokenAddress);
+
+        vm.prank(tokenGrantRecipient);
+        token.transfer(address(validator), INITIAL_BALANCE);
+
+        //deploy timelock
+        timelock = new Timelock(DELAY, proposers, executors, timelockAdmin);
+
+        DeployStakeTableScript stakeTableDeployer = new DeployStakeTableScript();
+        (proxy,) =
+            stakeTableDeployer.run(tokenAddress, address(lcMock), ESCROW_PERIOD, address(timelock));
+        stakeTable = StakeTableMock(proxy);
+    }
+
+    function test_initialize_sets_timelock_as_owner() public {
+        assertEq(stakeTable.owner(), address(timelock));
+    }
+
+    function test_timelock_upgrade_proposal_and_execution_succeeds() public {
+        vm.startPrank(proposers[0]);
+
+        // Encode upgrade call
+        bytes memory data =
+            abi.encodeWithSignature("upgradeToAndCall(address,bytes)", address(new S()), "");
+
+        bytes32 txId = timelock.hashOperation(address(stakeTable), 0, data, bytes32(0), bytes32(0));
+
+        timelock.schedule(address(stakeTable), 0, data, bytes32(0), bytes32(0), DELAY);
+
+        vm.stopPrank();
+
+        vm.assertFalse(timelock.isOperationReady(txId));
+
+        vm.warp(block.timestamp + DELAY + 1);
+
+        vm.assertTrue(timelock.isOperationReady(txId));
+
+        vm.startPrank(executors[0]);
+        timelock.execute(address(proxy), 0, data, bytes32(0), bytes32(0));
+        vm.stopPrank();
+        vm.assertTrue(timelock.isOperationDone(txId));
+    }
+
+    function test_expect_revert_when_timelock_is_not_owner() public {
```

Comment:
i thought the convention for case is `test_RevertWhen_TimelockIsNotOwner()`, as described in [pointer 6 here](https://book.getfoundry.sh/guides/best-practices#general-test-guidance)

Reply:
Thanks for pointing that out, i wasn't aware of the convention.

Reply:
tbh, i'm defintely not a fan of these `mixedNotSnake_NotCamel_WhimsicalLooking` suggestions, but i still tend to follow them... 🤦

Reply:
https://github.com/EspressoSystems/espresso-network/commit/a6d8d29351be37c97c34957fab18ff9e0adf0437